devtools/common/src/test/java/io/quarkus/cli/commands/CreateProjectTest.java

```diff
@@ -186,4 +193,60 @@ public static void delete(final File file) throws IOException {
         Assertions.assertFalse(
                 Files.exists(file.toPath()), "Directory still exists");
     }
+
+    @Test
+    public void createZip() throws IOException {
+        final File file = new File("target/zip");
+        delete(file);
+        file.mkdirs();
+        File zipFile = new File(file, "project.zip");
+        try (FileOutputStream fos = new FileOutputStream(zipFile);
+                ZipOutputStream zos = new ZipOutputStream(fos);
+                ZipProjectWriter zipWriter = new ZipProjectWriter(zos)) {
+            final CreateProject createProject = new CreateProject(zipWriter).groupId("io.quarkus")
+                    .artifactId("basic-rest")
+                    .version("1.0.0-SNAPSHOT");
+            Assertions.assertTrue(createProject.doCreateProject(new HashMap<>()));
+        }
+        Assertions.assertTrue(zipFile.exists());
+        File unzipProject = new File(file, "unzipProject");
+        try (FileInputStream fis = new FileInputStream(zipFile); ZipInputStream zis = new ZipInputStream(fis)) {
+            ZipEntry zipEntry = zis.getNextEntry();
+            byte[] buffer = new byte[1024];
+            while (zipEntry != null) {
+                File newFile = newFile(unzipProject, zipEntry);
+                if (zipEntry.isDirectory()) {
+                    newFile.mkdirs();
+                } else {
+                    new File(newFile.getParent()).mkdirs();
+                    FileOutputStream fos = new FileOutputStream(newFile);
+                    int len;
+                    while ((len = zis.read(buffer)) > 0) {
+                        fos.write(buffer, 0, len);
+                    }
+                    fos.close();
+                }
+                zipEntry = zis.getNextEntry();
+            }
+            zis.closeEntry();
+            zis.close();
```

Comment:
Super minor nitpick -> this is redundant. Feel free to ignore if you don't feel like removing it :)

Reply:
done